### PR TITLE
Implement Jacobian

### PR DESCRIPTION
**Pull request by certik**
@moorepants, this is what we need for PyDy. Now I am implementing the differentiation with respect to a function symbol.

### CODE REVIEW

File: symengine/python/symengine/lib/symengine_wrapper.pyx [hunk 1]
```diff
@@ -834,6 +834,14 @@ cdef class DenseMatrix(MatrixBase):
         deref(self.thisptr).FFLDU(deref(L.thisptr), deref(D.thisptr), deref(U.thisptr))
         return L, D, U
 
+    def jacobian(self, x):
+        cdef DenseMatrix x_ = sympify(x)
+        R = DenseMatrix(self.nrows(), x.nrows(), [0]*self.nrows()*self.nrows())
```
Comment (isuruf) [inline symengine/python/symengine/lib/symengine_wrapper.pyx]: This should be `[0]*self.nrows()*x.nrows()`

Comment (isuruf) [inline symengine/python/symengine/lib/symengine_wrapper.pyx]: Or simply `R = DenseMatrix(self.nrows(), x.nrows())`

Comment (isuruf) [inline symengine/python/symengine/lib/symengine_wrapper.pyx]: `R = DenseMatrix(self.nrows(), x.nrows())` doesn't work. I'll send a PR

Comment (certik) [inline symengine/python/symengine/lib/symengine_wrapper.pyx]: Yes, I was thinking about it, that we should allow just `DenseMatrix(self.nrows(), x.nrows())`. Thanks for the catch, I was rebasing before sending a PR and I think this bug cropped in during a rebase.

Comment (certik) [inline symengine/python/symengine/lib/symengine_wrapper.pyx]: I fixed it in 6b09b9e538f90af66c1815a933b25dfc1ede7041.

**Comment by certik** [comment]
I see, I got burned by enabling Debug mode but not the assertions. I am fixing it.

**Comment by certik** [comment]
Assertions enabled automatically in #528.

**Comment by certik** [comment]
@isuruf this now passes tests. Let me know if you are ok with this PR.

**Comment by isuruf** [comment]
+1 to merge

**Pull request merged**